File: app/src/main/java/com/jameskbride/eventbusmvpdemo/network/NetworkErrorViewFactory.java
package com.jameskbride.eventbusmvpdemo.network;

import android.os.Bundle;

import com.jameskbride.eventbusmvpdemo.bus.NetworkRequestEvent;

public class NetworkErrorViewFactory {
    private static final String NETWORK_REQUEST = "NETWORK_REQUEST";

    public NetworkErrorViewFragment make(NetworkRequestEvent networkRequestEvent) {
        Bundle bundle = new Bundle();
        bundle.putSerializable(NETWORK_REQUEST, networkRequestEvent);
        NetworkErrorViewFragment networkErrorViewFragment = new NetworkErrorViewFragment();
        networkErrorViewFragment.setArguments(bundle);

        return networkErrorViewFragment;
    }
}

File: app/src/main/java/com/jameskbride/eventbusmvpdemo/network/NetworkErrorViewFactory.kt
package com.jameskbride.eventbusmvpdemo.network

import android.os.Bundle
import com.jameskbride.eventbusmvpdemo.bus.NetworkRequestEvent
import com.jameskbride.eventbusmvpdemo.network.NetworkErrorViewFragment.Companion.NETWORK_REQUEST


class NetworkErrorViewFactory {

    fun make(networkRequestEvent: NetworkRequestEvent): NetworkErrorViewFragment {
        val bundle = Bundle()
        bundle.putSerializable(NETWORK_REQUEST, networkRequestEvent)
        val networkErrorViewFragment = NetworkErrorViewFragment()
        networkErrorViewFragment.arguments = bundle

        return networkErrorViewFragment
    }
}